buildSrc/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-testing.gradle.kts

```diff
@@ -0,0 +1,116 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+
+plugins {
+  `java-library`
+
+  id("net.bytebuddy.byte-buddy")
+
+  id("io.opentelemetry.instrumentation.base")
+  id("io.opentelemetry.instrumentation.javaagent-codegen")
+  id("io.opentelemetry.instrumentation.javaagent-shadowing")
+}
+
+dependencies {
+  // Integration tests may need to define custom instrumentation modules so we include the standard
+  // instrumentation infrastructure for testing too.
+  compileOnly("io.opentelemetry.instrumentation:opentelemetry-instrumentation-api")
```

Comment:
Compared to the previous version, these have been changed from project dependencies (which are subbed back in by otel.java-conventions)